File: datalib/src/main/java/nl/acidcats/tumblrlikes/core/repositories/LikesDataRepository.java
package nl.acidcats.tumblrlikes.core.repositories;

import java.util.List;

import nl.acidcats.tumblrlikes.core.models.Photo;
import rx.Observable;

/**
 * Created by stephan on 28/03/2017.
 */

public interface LikesDataRepository {
    Observable<List<Photo>> getLikedPhotos(String blogName, int count, long beforeTime);

    boolean hasMoreLikes(long mostRecentCheckTime);

    long getLastLikeTime();
}

File: datalib/src/main/java/nl/acidcats/tumblrlikes/core/repositories/LikesDataRepository.kt
package nl.acidcats.tumblrlikes.core.repositories

import nl.acidcats.tumblrlikes.core.models.Photo
import rx.Observable

/**
 * Created on 24/10/2018.
 */
interface LikesDataRepository {
    fun getLikedPhotos(blogName: String, count: Int, beforeTime: Long): Observable<List<Photo>>

    fun isLoadComplete(): Boolean

    fun getLastLikeTime(): Long
}